### ISSUE TITLE
Versionbot creates quite a bit of extra noise in the git history

### ISSUE THREAD

**Issue by pimterry** [resin-io-modules/resin-procbots#48]
Personal opinion, so this may be contentious, but versionbot adds quite a few extra commits to the history (if you were doing commit -> merge, it's now commit -> version -> merge, so 50% extra), and that does obscure the actual changes a little.

Having added versionbot to resin-ui for example, the git history is significantly less clear than it was before. Compare Resin UI's [current history](https://github.com/resin-io/resin-ui/commits/master) with its [previous history](https://github.com/resin-io/resin-ui/commits/master?after=Y3Vyc29yOisU9V5jDN6vRNTOTdskb6wHfUQbKzY5).

Could we bundle the version update and the merge into a single commit? Do other people not mind at all? Are there any other options?

**Comment by LucianBuzzo** [comment]
Bundling the version update and merge into a single commit would be great. 
However, I don't mind the versionbot output as it is at the moment because even though it adds more commits, all the commits are consistent and uniform.

**Comment by hedss** [comment]
I don't disagree at all with this. I guess we need a `git merge --squash` (or the raw data handling of it, as that's how VB has to operate on the repo) just before the tagging.

**Comment by lurch** [comment]
VisualBasic?!? :rofl:  :wink:

**Comment by pimterry** [comment]
`git merge --squash` would squash down the entire branch history to a single merge commit, right? Most of the time the more detailed history of actual changes is useful, imo - it's just the final merge + version update I'd like to combine, not the whole of the PR's history.

Shouldn't be too hard using git directly, but I'm not sure how easy that is to do that with the github api.

**Comment by lurch** [comment]
OTOH with the "current history" link I guess it might be useful seeing exactly when a given version was released, which I guess will become more significant when #16 comes into play.
So if this "reduced history noise" comes into effect, I guess it'd be useful if VersionBot also edited the commit/merge title to show the new version number (when applicable)?
(or perhaps what I'm suggesting is irrelevant, because tags would be more useful for seeing that anyway?)

**Comment by hedss** [comment]
Yeah, I think it should be doable as you can do *most* of the underlying reference handling operations with the Github API. But, this is a caveat as this, yet again, might be something missing. :-/

Agree wrt to history, I was thinking of a squash where the last commit (ie. the version up) was merged with the merge commit itself.

@lurch The merge commit should give you this, unless I'm missing something?

**Comment by lekkas** [comment]
@hedss have we considered rebase merges?

**Comment by hedss** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [VersionBot: Creates quite a bit of extra noise in the git history](https://trello.com/c/7a3rkgNy/6-versionbot-creates-quite-a-bit-of-extra-noise-in-the-git-history)
